service/controller/chart/v1/resource/release/create.go

```diff
@@ -36,7 +36,26 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
 
 		tarballPath, err := r.helmClient.PullChartTarball(ctx, tarballURL)
 		if helmclient.IsPullChartFailedError(err) {
-			r.logger.LogCtx(ctx, "level", "warning", "message", "pulling chart failed", "stack", microerror.Stack(err))
+			reason := fmt.Sprintf("pulling chart %#q failed", tarballURL)
```

Comment:
Since we don't pull the chart tarball in the desired state we need more error handling here.